### PR TITLE
Refactor unit testing

### PR DESCRIPTION
**Pull request by JellevanKraaij**
**Testing**
- improve naming by adding prefix to Cmake
- create test fixture for a basic window

**Workflow**
- removed workflow_dispatch as it should :pray: not change anything on its own

### CODE REVIEW

**Comment by JellevanKraaij** [comment]
Did I totally break the workflow by removing workflow_dispatch?
Maybe the settings of when a workflow is allowed to run is changed? @W2Wizard maybe you need to approve it?
Otherwise, I guess the workflow_dispatch is somehow bugging the run... and i will change it back in that case :confused:

**Comment by W2Wizard** [comment]
Lol, it just doesn't run ? Usually the `workflow_dispatch` is for manual execution.

File: tests/WindowFixture.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# include <gtest/gtest.h>
+
+# include <MLX42/MLX42.h>
+
+class Window : public ::testing::Test
+{
+protected:
+	mlx_t* mlx = NULL;
+
+	static constexpr const char *name = "MLX42";
```
Comment (W2Wizard) [inline tests/WindowFixture.hpp:12]: Uber cringe detected: `*` not with the type. 🤣

Comment (JellevanKraaij) [reply tests/WindowFixture.hpp:12]: okay okay okay
i will change my style

**Comment by JellevanKraaij** [comment]
the workflow dispatch should not make a difference

**Comment by W2Wizard** [comment]
> the workflow dispatch should not make a difference

Yeah makes no sense, but it's not even detecting the workflow thing to begin with 😕

**Review by W2Wizard — APPROVED** [review]

**Comment by JellevanKraaij** [comment]
The stupid CI is always working in stupid ways

**Comment by JellevanKraaij** [comment]
Just to be sure, its not my PR

**Comment by JellevanKraaij** [comment]
https://www.githubstatus.com/incidents/qfby3shb6r3w

**Comment by W2Wizard** [comment]
> https://www.githubstatus.com/incidents/qfby3shb6r3w guess GitHub made a small error

RIP to the intern who broke it 😢

**Pull request merged**